twilight-gateway/src/future.rs

```diff
@@ -56,6 +56,8 @@ pub struct NextMessageFuture<'a> {
     channel_receive_future: &'a mut UnboundedReceiver<Message>,
     /// Future resolving when the next Websocket message has been received.
     message_future: Next<'a, Connection>,
+    /// Command ratelimiter, if enabled.
+    maybe_ratelimiter: Option<&'a mut CommandRatelimiter>,
```

Comment:
we could update the docs for the `NextMessageFuture` type to indicate that it only relays messages when the ratelimiter has permits available. i couldn't target it for comments though, which is why i'm posting here